endpoints/openrtb2/auction.go

```diff
@@ -314,6 +320,68 @@ func parseTimeout(requestJson []byte, defaultTimeout time.Duration) time.Duratio
 	return defaultTimeout
 }
 
+// mergeBidderParams merges bidder parameters passed at req.ext level with imp[].ext level.
+// Preference is given to parameters at imp[].ext level over req.ext level.
+// Parameters at req.ext level are propagated to adapters as is without any validation.
+func mergeBidderParams(req *openrtb_ext.RequestWrapper) error {
+	var reqBidderParams map[string]json.RawMessage
+
+	reqBidderParams, err := adapters.ExtractBidderParams(req.BidRequest)
+	if err != nil {
+		return err
+	}
+
+	impCpy := make([]openrtb2.Imp, 0)
+	for _, imp := range req.BidRequest.Imp {
+		updatedImp := imp
+		var impExt map[string]map[string]interface{}
```

Comment:
Nitpick: Please move the variable definition closer to it's variable usage by the Unmarshal call.

Reply:
Fixed.